src/main/java/seedu/address/logic/parser/EditSupplierCommandParser.java

```diff
@@ -81,6 +98,57 @@ public EditSupplierCommand parse(String args) throws ParseException {
         return new EditSupplierCommand(name, editSupplierDescriptor);
     }
 
+    /**
+     * Checks all invalid fields for supplier class.
+     * @param fieldArgMultimap The prefix to value mapping.
+     * @throws ParseException Throws when there exist invalid fields.
+     */
+    public void checkEditFieldValidity(ArgumentMultimap fieldArgMultimap) throws ParseException {
+        ArrayList<String> invalidFields = new ArrayList<>();
+        if (arePrefixesPresent(fieldArgMultimap, PREFIX_NAME)) {
+            invalidFields.add("name");
+        }
+        if (arePrefixesPresent(fieldArgMultimap, PREFIX_SALARY)) {
+            invalidFields.add("salary");
+        }
+        if (arePrefixesPresent(fieldArgMultimap, PREFIX_EMPLOYMENT)) {
+            invalidFields.add("employment");
+        }
+        if (arePrefixesPresent(fieldArgMultimap, PREFIX_SKILL)) {
+            invalidFields.add("skill");
+        }
+        if (arePrefixesPresent(fieldArgMultimap, PREFIX_COMMISSION)) {
+            invalidFields.add("commission");
+        }
+        if (invalidFields.size() > 0) {
+            throw new ParseException(formatInvalidFieldString(invalidFields));
+        }
```

Comment:
Good catch of all error before throwing ParseException. This is a very detailed checking to help user to know which field should not exist in current edit